tools/wt_verify/wt_verify.py

```diff
@@ -172,6 +237,129 @@ def parse_output():
     return output
 
 
+def show_page_distribution(filename, data, bar_width=1):
+    """
+    data: dictionary generated by parse_dump_blocks() that contains information related to different
+    checkpoints.
+    """
+    imgs = ""
+    bar_width = 1
+    colors = ['blue', 'orange', 'green', 'red', 'purple', 'brown', 'pink', 'gray', 'olive', 'cyan']
+    for checkpoint in data:
+
+        fig, ax = plt.subplots(1, figsize=(15, 10))
+        max_addr = 0
+        max_addr_size = 0
+
+        for i, key in enumerate(data[checkpoint]):
+            ax.broken_barh(data[checkpoint][key], (i, bar_width),
+                           facecolors=f'tab:{colors[i]}', label=key)
+
+            # Save the max offset seen for the plot assuming the offsets are sorted.
+            if data[checkpoint][key][-1][0] > max_addr:
+                max_addr = data[checkpoint][key][-1][0]
+                max_addr_size = data[checkpoint][key][-1][1]
+
+        # Plot settings.
+        plt.yticks(np.arange(0, len(data[checkpoint]) + 1, 1))
+        xlimit = max_addr + max_addr_size
+        plt.xlim(left=0,right=xlimit)
+        plt.xlabel("Offset")
+        plt.ylabel("Page type")
+        plt.legend()
+        plt.title(f"Distribution of each page type for {filename} - {checkpoint}")
+        plt.close()
+
+        img = mpld3.fig_to_html(fig)
+        imgs += img
+    return imgs
+
+def show_block_distribution(filename, data, bins=100):
+    """
+    data: dictionary generated by parse_dump_blocks() that contains information related to different
+    checkpoints.
+    """
+    imgs = ""
+    for checkpoint in data:
+        all_addr = []
+
+        # Concatenate the lists for each page type:
+        for page_type in data[checkpoint]:
+            all_addr += data[checkpoint][page_type]
+
+        # Create a list of offsets.
+        all_addr = list(map(itemgetter(0), all_addr))
+
+        fig, ax = plt.subplots(1, figsize=(15, 10))
+        ax.hist(all_addr, bins=bins)
+        plt.title(f"Distribution of blocks for {filename} - {checkpoint}")
+        plt.xlabel(f"Offset (bins={bins})")
+        plt.ylabel("Frequency")
+        plt.close()
+
+        img = mpld3.fig_to_html(fig)
+        imgs += img
+    return imgs
+
+
+def show_free_block_distribution(filename, data, max_gap_size=0):
+    """
+    data: dictionary generated by parse_dump_blocks() that contains information related to different
+    checkpoints.
+    max_gap_size: display free blocks up to the specified size in bytes
+    """
+    imgs = ""
+    for checkpoint in data:
+        all_addr = []
+
+        # Concatenate the lists for each page type:
+        for page_type in data[checkpoint]:
+            all_addr += data[checkpoint][page_type]
+
+        # Sort them by the addr_start which is the first element:
+        all_addr.sort(key=itemgetter(0))
+
+        # Now count all the gaps:
+        buckets = {}
+        for i, current_tupe in enumerate(all_addr):
```

Comment:
Possible typo - is `tupe` meant to be `tuple` or `tup`?

Reply:
Nice catch! Renamed in 1900f9bf581c875cdd293fcdbe5a4fec04ddda87.

Reply:
Thanks!